src/socket/dns.rs

```diff
@@ -81,6 +90,7 @@ struct PendingQuery {
     delay: Duration,
 
     server_idx: usize,
+    mdns: bool,
```

Comment:
This could be a "Protocol" enum, with valus "Dns" and "Mdns".

Reply:
I don't think the API should encourage thinking about mDNS and DNS queries as fundamentally different protocols

Reply:
I did something in this vein when I put this behind a feature flag